yt/utilities/linear_interpolators.py

```diff
@@ -1,11 +1,82 @@
+import abc
+
 import numpy as np
 
 import yt.utilities.lib.interpolators as lib
 from yt.funcs import mylog
 
 
-class UnilinearFieldInterpolator:
-    def __init__(self, table, boundaries, field_names, truncate=False):
+class _LinearInterpolator(abc.ABC):
+    _ndim: int
+    _dim_i_type = "int32"
+
+    def __init__(self, table, field_names, truncate=False, *, store_table=True):
+        if store_table:
+            self.table = table.astype("float64")
+        else:
+            self.table = None
+        self.table_shape = table.shape
+        self.truncate = truncate
+        self._field_names = field_names
+
+    def _raise_truncation_error(self, data_object):
+        mylog.error(
+            "Sorry, but your values are outside "
+            "the table!  Dunno what to do, so dying."
+        )
+        mylog.error("Error was in: %s", data_object)
+        raise ValueError
```

Comment:
I'm not sure I see the value of having log errors *and* an exception. In my opinion, the exception is where information should be given, making log errors redundant.

Reply:
yup, I agree, I was again just copying exactly what was here before. In this case though I think it's clear that there's not reason not to change this to simply raise the error.

Reply:
ok, changed this line as well as other spots throughout that were logging and raising errors.